iree/base/dynamic_library_test.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/base/dynamic_library.h"
+
+#include <string>
+
+#include "iree/base/dynamic_library_test_library_embed.h"
+#include "iree/base/file_io.h"
+#include "iree/base/status.h"
+#include "iree/base/status_matchers.h"
+#include "iree/base/target_platform.h"
+#include "iree/testing/gtest.h"
+
+namespace iree {
+namespace {
+
+static const char* kUnknownName = "library_that_does_not_exist.so";
+
+class DynamicLibraryTest : public ::testing::Test {
+ public:
+  static void SetUpTestCase() {
+    std::string base_name = "dynamic_library_test_library";
+    ASSERT_OK_AND_ASSIGN(library_temp_path_, file_io::GetTempFile(base_name));
```

Comment:
So I know this isn't what this PR is about, but reading this now, I'm confused why we named this function `GetTempFile` instead of `GetTempFilename` or `MakeTempFilename`. It doesn't return a filehandle; it returns a path. It looks especially weird here because then you're modifying the returned name.

Reply:
I saw some variations across temp file implementations. Some seemed to create a file and return its path while others would just return a path without creating the file yet. One of my references was this code we used on a prior project: https://github.com/google/xrtl/blob/master/xrtl/testing/file_util.h which did actually return file handles.

We could rename to `MakeTempFilename` or some similar option.

Reply:
Yeah I'd vote for that, but not this PR :-)